### PR TITLE
0 -> `zero(step)`

### PR DESCRIPTION
**Pull request by longemen3000**
this package is the exact solution for this discourse problem:
https://discourse.julialang.org/t/open-ended-unitrange/102287/5
but it does not support comparisons when the step is a `Dates.Period`. this PR tries to fix comparisons between the step and `0`

### CODE REVIEW

**Comment by jw3126** [comment]
Awesome! Can you add a test case with dates?

**Comment by longemen3000** [comment]
of course, one minute

**Comment by longemen3000** [comment]
before merging, let me test the cases where probably `oneunit(step)` is needed

**Comment by longemen3000** [comment]
i was, at first, worried for `floatlength`, but it seems to work fine with date ranges

**Comment by longemen3000** [comment]
Hmmm, tests fail with 1.0, because there are missing definitions on the dates module, it is ok to add a condition on the version there?

**Comment by jw3126** [comment]
> Hmmm, tests fail with 1.0, because there are missing definitions on the dates module, it is ok to add a condition on the version there?

You can bump to 1.6, we don't need to support 1.0 anymore.

File: Project.toml [hunk 1]
```diff
@@ -10,8 +10,9 @@ LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 julia = "1"
```
Comment (jw3126) [inline Project.toml]:
```suggestion
julia = "1.6"
```

**Comment by longemen3000** [comment]
Weird, there is a change in the Dates stdlib that allows this to work on 1.9 but not on 1.6

**Comment by jw3126** [comment]
Is it hard to make it work on 1.6?

**Comment by jw3126** [comment]
If it is a headache to fix, you can just put a `if VERSION >= v"1.9"` around the test.

**Comment by jw3126** [comment]
Thanks @longemen3000 this is a very nice feature.

**Pull request merged**